### PR TITLE
[feature] Clicks tracking by device type and country and pie charts

### PR DESCRIPTION
**Pull request by larrabee**
Hello. 
Features:

- Device type tracking for opened emails and clicked links
- Pie charts with stats by country/device type for opened emails and clicked links
  
Images:

- [Screenshot. Opened emails](http://imgur.com/0t7KTnr)
- [Screenshot. Link info](http://imgur.com/haAFXMv)

### CODE REVIEW

File: views/campaigns/clicked.hbs [hunk 1]
```diff
@@ -69,6 +69,31 @@
         <!-- Default panel contents -->
         <div class="panel-heading">{{#if aggregated}}{{#translate}}Subscribers who clicked on a link:{{/translate}}{{else}}{{#translate}}Subscribers who clicked on this link:{{/translate}}{{/if}}</div>
         <div class="panel-body">
+          {{#if aggregated}}{{else}}
```
Comment (witzig) [inline views/campaigns/clicked.hbs]: Can we use  `{{#unless aggregated}}` here?

Comment (larrabee) [inline views/campaigns/clicked.hbs]: Fixed

File: lib/models/campaigns.js [hunk 2]
```diff
@@ -193,7 +193,43 @@ module.exports.filterClickedSubscribers = (campaign, linkId, request, columns, c
...
+};
 
+module.exports.statsClickedSubscribersByColumn = (campaign, linkId, request, column, limit, callback) => {
+    db.getConnection((err, connection) => {
+        if (err) {
+            return callback(err);
+        }
+
+        let values = [campaign.list, linkId];
+        let query = 'SELECT SQL_CALC_FOUND_ROWS ' + column + ' AS data, COUNT(*) AS cnt FROM `subscription__' + campaign.list + '` JOIN `campaign_tracker__' + campaign.id + '` ON `campaign_tracker__' + campaign.id + '`.`list`=? AND `campaign_tracker__' + campaign.id + '`.`subscriber`=`subscription__' + campaign.list + '`.`id` AND `campaign_tracker__' + campaign.id + '`.`link`=? GROUP BY ' + column + ' ORDER BY COUNT(' + column + ') DESC,' + column;
```
Comment (witzig) [inline lib/models/campaigns.js]: I'd prefer this to be formatted a bit.

**Comment by witzig** [comment]
This would be a lovely addition to Mailtrain, IMHO.

May I suggest you run `eslint:all` and fix the formatting? It'll make it easier for Andris to review. Just run `grunt` from command line.

**Comment by larrabee** [comment]
Formating fixed, replace query string concatenations with util.format().

**Comment by andris9** [comment]
Could you upgrade your branch from the master as there are new changes and fix any potential conflicts? Additionally, use 23 instead of 22 for the mysql upgrade identifier schema version as the 22 is already taken. Otherwise seems nice.

**Comment by andris9** [comment]
Thanks!

**Comment by larrabee** [comment]
You too.

**Comment by err0r3o3** [comment]
Would be great to be able to disable this tracking feature.
Is there an option to do that?

**Pull request merged**